### ISSUE TITLE
oxfmt: non-idempotent formatting of a wrapping second paragraph in a markdown list item (regression in 0.58.0)

### ISSUE THREAD

**Issue by bensandee** [oxc-project/oxc#25000]
## Summary

When a markdown list item contains a **second paragraph** whose text wraps, oxfmt adds 4 spaces
to that paragraph's continuation lines on every pass. The output never converges, so
`oxfmt --check` can never pass and running `oxfmt` to fix it grows the file again.

This is a regression: 0.57.0 is stable, 0.58.0 is not.

## Reproduction

No config file, no `.editorconfig` -- stock defaults in an empty directory.

`t.md`:

```markdown
- [x] **Lead.** A first paragraph inside the list item, long enough that the formatter has to
      wrap it across more than one line to respect the configured print width.

      A second paragraph inside the same list item, also long enough that the formatter has to
      wrap it across more than one physical line of output text here.
```

Run `oxfmt t.md` three times, measuring the maximum leading-space count:

```
pass 1: max-indent=10
pass 2: max-indent=14
pass 3: max-indent=18
```

Each pass adds 4 spaces to the second paragraph's continuation line, without bound. The first
paragraph's continuation line is unaffected and stays at 6.

## Expected

Formatting is idempotent: `oxfmt` on already-formatted output produces no change, and
`oxfmt --check` passes on the output of `oxfmt`.

## Actual

`oxfmt --check` fails on the output of a single `oxfmt` pass, and keeps failing after any
number of passes, because each pass produces a new, larger file.

## Bisect

Same file, same stock defaults, three passes each:

| Version | max-indent per pass | Verdict |
| ------- | ------------------- | ------- |
| 0.45.0  | 10 / 10 / 10        | stable  |
| 0.51.0  | 10 / 10 / 10        | stable  |
| 0.55.0  | 10 / 10 / 10        | stable  |
| 0.56.0  | 10 / 10 / 10        | stable  |
| 0.57.0  | 6 / 6 / 6           | stable  |
| 0.58.0  | 10 / 14 / 18        | **grows** |
| 0.59.0  | grows               | grows   |
| 0.60.0  | grows               | grows   |
| 0.61.0  | grows               | grows   |

**The regression is introduced in 0.58.0.** 0.61.0 (current latest) does not fix it.

To rule out the older versions merely ignoring markdown: 0.55.0 was checked against a different
unformatted markdown construct and did reformat it, so markdown support is present and these
versions are genuinely stable on this construct rather than inert.

## Relationship to existing issues

This is not covered by either existing non-idempotency report:

- #21314 -- HTML comment as a list continuation, +2 spaces/pass, reported on 0.44.0. Closed as
  not planned.
- #20778 -- fenced code blocks in lists, reported on 0.42.0, attributed to Prettier
  (prettier/prettier#10063, open since 2021). Closed as not planned.

This case has **no HTML comment and no code fence** -- it is a plain paragraph. More
importantly, it is a **regression with a known version boundary** (0.57.0 clean, 0.58.0 broken)
rather than longstanding upstream Prettier behavior, so whatever changed in 0.58.0 is the
likely cause and the bisect should localize it.

## Impact

The construct is common in task lists and changelogs: a checkbox item followed by an
explanatory paragraph. Any repository containing it cannot pass a `--check` gate, and CI setups
that run oxfmt in write mode and commit the result will produce a new commit on every run.

## Environment

- oxfmt 0.58.0 through 0.61.0 (0.57.0 unaffected)
- Linux x86_64, Node 24
- No `.oxfmtrc.json`, no `.editorconfig` -- stock defaults

**Comment by leaysgur** [comment]
- https://github.com/prettier/prettier/issues/19644
- https://github.com/prettier/prettier/pull/19647

However, I'm not sure when the next release from Prettier is.

**Comment by jpease** [comment]
Following up with measurements on the open question here ("I'm not sure when the next release from Prettier is"), plus one case that raises the severity.

## The Prettier fix is in the 4.0 alpha line, and in no released 3.x

Verified directly rather than inferred:

| build | this issue's exact repro |
| --- | --- |
| `prettier@3.9.6` (npm `latest`) | **diverges** — +4/pass, unbounded |
| `prettier@4.0.0-alpha.13` (npm `next`) | **converges** |

So #19647 is merged but has never shipped in a 3.x release — its changelog entry is still under `changelog_unreleased/markdown/19647.md`. Anyone on `prettier@latest` still reproduces it today.

## It still reproduces on oxfmt 0.63.0

Using the exact `t.md` from this issue's body, stock defaults, empty directory, measuring max leading-space count per pass:

```
oxfmt 0.57.0:  pass1=6   pass2=6   pass3=6     <- stable, as reported
oxfmt 0.61.0:  pass1=10  pass2=14  pass3=18
oxfmt 0.63.0:  pass1=10  pass2=14  pass3=18    <- latest
```

0.63.0 was published 2026-08-10, thirteen days after this issue was closed, so the closure is still ahead of the behaviour. Not asking for a reopen if the duplicate framing is how you prefer to track it — flagging it because the trail currently ends at a closed issue whose repro is live in the newest release, and the Prettier issue it points to is also closed.

## A case not in the original report: the fenced-code variant is corruption, not churn

The reported symptom is unstable indentation. With a fenced code block as the continuation, it also destroys the block's structure — the **opening** fence keeps its indent while the content and the **closing** fence both drift:

```
- [ ] Item text here.

      ```sh
      echo hi
      ```
```

after three passes:

```
- [ ] Item text here.

      ```sh
                  echo hi
                  ```
```

Once the closing fence sits more than three spaces deeper than the opener, CommonMark no longer recognises it as a closing fence, so the code block stops being terminated and the rest of the document is swallowed into it. That is a content change, not a formatting one. It converges correctly on `prettier@4.0.0-alpha.13`, so #19647 covers this case too — worth knowing it is covered, since the fixture on that PR only exercises the wrapped-paragraph shape.

Measured across the same matrix: `- [ ]` and `- [x]` behave identically, a plain `- ` bullet with the same continuation converges fine, and a task item with only **one** continuation block is also stable — it is content past the first continuation block that drifts. Config is not a factor: reproduces with no `.oxfmtrc.json` discoverable at all, and with every individual option set.

## Workaround, for anyone who lands here

`<!-- prettier-ignore -->` immediately before the task list is honoured by oxfmt and makes the file `--check`-clean and byte-stable. `<!-- oxfmt-ignore -->` is not honoured. Also note a fence tagged ` ```markdown ` is formatted recursively, so quoting the repro reproduces the bug in the quoting document — use an untagged or `text` fence.
